### DISCUSSION TITLE
GTK: headless compositor tests

### DISCUSSION THREAD

**Discussion by jcollie** [ghostty-org/ghostty#2359] [Feature Requests, Ideas]
There's a lot of GTK code that is essentially untested. Using a headless compositor running in a docker container we can do
some basic checks to make sure that Ghostty launches and that GTK isn't issuing any warnings.

This StackExchange answer provides a starting point:

https://unix.stackexchange.com/a/735032

Checking the Gnome CI system for other hints would be useful too.

**Comment by jcollie** [comment]
Just realized that we'll need a software OpenGL driver or run this on a CI instance that has GPU access.

**Comment by mitchellh** [comment]
No GPU is required, we can use a software driver, they're plenty fast enough.
